### ISSUE TITLE
Play does not closes the connection upon InputStream error

### ISSUE THREAD

**Issue by ccarezza** [playframework/playframework#5269]
When Play 2.4.3 is serving a chunked response from a Java InputStream, if the read operation throws an exception, the connection to the browser is never closed.

The following Java code reproduces the issue:
```java
public Result serveInputStream() {
    return ok(new InputStream() {
        @Override
        public int read() throws IOException {
            throw new IOException();
        }
    });
}
```

**Comment by mkurz** [comment]
@cchantep Is this still an issue with Play 2.5?

**Comment by mkurz** [comment]
I can reproduce this with Play 2.4.6.
But I can **not** reproduce this with Play 2.5 anymore (using the exact same code, trying a couple of times).

I consider this issue as fixed and therefore close it.
If someone thinks I am wrong or thinks this has to be fixed in the 2.4.x branch feel free to reopen.

**Comment by ccarezza** [comment]
I've tested in play 2.5 and it is working as it should accordingly to HTTP 1.1 specification.

Thanks for fixing it.
